packages/flutter/lib/src/cupertino/nav_bar.dart

```diff
@@ -697,12 +713,16 @@ class CupertinoSliverNavigationBar extends StatefulWidget {
     this.transitionBetweenRoutes = true,
     this.heroTag = _defaultHeroTag,
     this.stretch = false,
+    this.drawer,
+    this.drawerMode = NavigationDrawerMode.none,
+    this.drawerHeight = _kNavBarDrawerHeight,
```

Comment:
I wonder if there's a way to combine these into one top level property, like how things like `Border` work. But that could be tricky with how we treat composability.

Reply:
We chatted offline today about looking into making drawer a PReferredSizeWidget, like AppBar.bottom. That could eliminate a property here in specifying the height, and would solve the conundrum of size before layout.

Reply:
Thanks @Piinks and @nate-thegrate for the PreferredSizeWidget suggestion!